mailscanner/user_manager.php

```diff
@@ -536,10 +563,17 @@ function change_state(id, filter) {
                 echo ' <TR><TH>' . __('filter12') . '</TH><TH>' . __('active12') . '</TH><TH>' . __('action12') . '</TH></TR>' . "\n";
                 if ($result->num_rows > 0) {
                     while ($row = $result->fetch_object()) {
-                        echo ' <TR><TD>' . $row->filter . '</TD><TD>' . $row->active . '</TD><TD>' . $row->actions . '</TD></TR>' . "\n";
+                        echo ' <TR><TD>' . $row->filter . '</TD><TD>' . $row->active . '</TD> ';
+                        if ($_SESSION['user_type'] === 'D' && $id === $_SESSION['myusername']) {
+                            echo '</TR>' . "\n";
+                        } else {
+                            echo '<TD>' . $row->actions . '</TD></TR>' . "\n";
+                        }
                     }
                 }
-                echo ' <TR><TD><INPUT TYPE="text" NAME="filter"></TD><TD><SELECT NAME="active"><OPTION VALUE="Y">' . __('yes12') . '<OPTION VALUE="N">' . __('no12') . '</SELECT></TD><TD><INPUT TYPE="submit" VALUE="' . __('add12') . '"></TD></TR>' . "\n";
+                if ($_SESSION['user_type'] !== 'D' || ($_SESSION['user_type'] === 'D' && $id !== $_SESSION['myusername'])) {
```

Comment:
I suggest to use `$_SESSION['user_type'] === 'A'` instead of `$_SESSION['user_type'] !== 'D'` for better understanding of the code and to print a message if  a domain admin tries to edit his own filter that he is not allowed to do